lib/wasix/src/state/env.rs

```diff
@@ -155,7 +155,13 @@ impl WasiInstanceHandles {
             signal: instance
                 .exports
                 .get_typed_function(&store, "__wasm_signal")
-                .ok(),
+                .ok()
+                .or_else(|| {
+                    instance
```

Comment:
Can we add a comment adding context and describing why this is happening?

Reply:
ping @john-sharratt